src/Terminal/Terminal.ts

```diff
@@ -102,7 +102,7 @@ export class Terminal {
     if (this.action === null) return;
     this.action.timeLeft -= (CONSTANTS.MilliPerCycle * cycles) / 1000;
     if (this.action.timeLeft < 0.01) this.finishAction(false);
-    TerminalEvents.emit();
+    TerminalProcessEvents.emit();
```

Comment:
```suggestion
    else TerminalProcessEvents.emit();
```
Only need to send a process event if there is still an action underway. Without the debounce (see other suggestion), sending the event anyway leads to the action timer trying to rerender when there's no action.